pageserver/src/layered_repository.rs

```diff
@@ -1743,8 +1758,33 @@ impl LayeredTimeline {
                 }
                 continue 'outer;
             }
-
-            // 2. Is it needed by a child branch?
+            // 2. It is newer than PiTR interval?
+            // We use modification time of layer file to estimate update time.
+            // This estimation is not quite precise but maintaining LSN->timestamp map seems to be overkill.
+            // It is not expected that users will need high precision here. And this estimation
+            // is conservative: modification time of file is always newer than actual time of version
+            // creation. So it is safe for users.
+            //
+            if let Ok(metadata) = fs::metadata(&l.filename()) {
+                let last_modified = metadata.modified()?;
```

Comment:
I am subtracting system times at pageserver. It is not somehow related with time at client side and time at S3. So even if time at pageserver is not correct, it should not cause problems, Certainly there can be cases when system time is adjusted. Either explicitly by admin, either implicitly by NTP protocol. In this case such side effects are really possible. But I do not expect that PiTR interval wil be specified with high resolution (seconds or even hours). Most probably it will be days or even months. In this case such time correction will not be critical.

Reply:
You're right, I think I thought more about TimelineMetadata here, but it is unrelated to this code, remaining concerns are covered here https://github.com/zenithdb/zenith/pull/1332#discussion_r823006105